core/src/main/scala/akka/persistence/r2dbc/internal/BySliceQuery.scala

```diff
@@ -222,6 +222,21 @@ import org.slf4j.Logger
         fromTimestamp: Instant,
         limit: Int): Future[Seq[Bucket]]
 
+    protected def appendTwoEmptyBucketsIfMissing(
+        buckets: IndexedSeq[Bucket],
+        toTimestamp: Instant): IndexedSeq[Bucket] = {
+      import Buckets.BucketDurationSeconds
+      val startTimeOfLastBucket = (toTimestamp.getEpochSecond / 10) * 10
+      val startTimeOf2ndLastBucket = ((toTimestamp.getEpochSecond - BucketDurationSeconds) / 10) * 10
```

Comment:
There could be more cases that are not covered by this, such as that last bucket is found but not 2nd last.

nextStartTime that was added in https://github.com/akka/akka-persistence-r2dbc/pull/641 is using the 2nd last, but it can be simplified by using the last - BucketDurationSeconds instead. Then I can just add the last bucket here if that is missing.

Reply:
https://github.com/akka/akka-persistence-r2dbc/pull/645/commits/77a6e0c39fcb72dedabee783ca65307f7a8b6bcd